services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -195,6 +195,44 @@ const indexBlock = async job => {
 					}, dbTrx);
 				}
 			}
+
+			// TODO: Verify and enable it once pos:validatorStaked schema is exposed from SDK
+			// Calculate locked amount change and update in key_value_store table for affected tokens
+
+			// Move this require to top
+			// const { KV_STORE_KEYS } = require('../constants');
+			// const EVENT_NAMES = Object.freeze({
+			// 	LOCKED: 'locked',
+			// 	UNLOCKED: 'unlocked',
+			// });
+
+			// const tokenIDLockedAmountChangeMap = {};
+			// events.forEach(event => {
+			// 	const { data: eventData } = event;
+			// 	// Initialize map entry with BigInt
+			// 	if ([EVENT_NAMES.LOCKED, EVENT_NAMES.UNLOCKED].includes(event.name)
+			// 		&& !(eventData.tokenID in tokenIDLockedAmountChangeMap[eventData.tokenID])) {
+			// 		tokenIDLockedAmountChangeMap[eventData.tokenID] = BigInt(0);
+			// 	}
+
+			// 	if (event.name === EVENT_NAMES.LOCKED) {
+			// 		tokenIDLockedAmountChangeMap[eventData.tokenID] += eventData.amount;
+			// 	} else if (event.name === EVENT_NAMES.UNLOCKED) {
+			// 		tokenIDLockedAmountChangeMap[eventData.tokenID] -= eventData.amount;
+			// 	}
+			// });
+
+			// await BluebirdPromise.map(
```

Comment:
Could you also please include the reverse logic in the deleteBlock job?

Reply:
I would prefer keep it plural.

Reply:
in progress. will need to update events fetching when suggested by SDK.

Reply:
This will be addressed in a separate issue: https://github.com/LiskHQ/lisk-service/issues/1440